apps/arweave/src/ar_node_worker.erl

```diff
@@ -2055,12 +2140,29 @@ handle_found_solution(Args, PrevB, State, IsRebase) ->
 						{false, rebase_threshold}
 				end
 		end,
-	%% Check steps and step checkpoints.
-	HaveSteps =
+
+	PrevCDiff = PrevB#block.cumulative_diff,
+	CDiff = ar_difficulty:next_cumulative_diff(PrevCDiff, Diff, Height),
+	NoDoubleSigning =
```

Comment:
We're adding this check here so that a miner can detect accidental double signing before they publish?

Reply:
Yes